vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExternalLinkManager.java

```diff
@@ -98,7 +120,32 @@ protected void removeFromLinkList(ISpace space, IExternalLink link) {
 
     @Override
     protected void deleteLinkRepo(IExternalLink link) {
-        externalLinkRepo.delete((ExternalLink)link);
+        externalLinkRepo.delete((ExternalLink) link);
+    }
+
+    @Override
+    public IExternalLinkDisplay createLink(String title, String id, float positionX, float positionY, int rotation,
+            String linkedId, String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename,
+            String howToOpen)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+
+        IExternalLinkDisplay externalLinkDisplay = createLink(title, id, positionX, positionY, rotation, linkedId,
+                linkLabel, displayType, linkImage, imageFilename);
+        externalLinkDisplay.setHowToOpen(howToOpen);
+        return externalLinkDisplay;
+
+    }
+
+    @Override
+    public IExternalLinkDisplay updateLink(String title, String id, float positionX, float positionY, int rotation,
+            String linkedId, String linkLabel, String linkId, String linkDisplayId, DisplayType displayType,
+            byte[] linkImage, String imageFilename, String howToOpen)
+            throws SpaceDoesNotExistException, LinkDoesNotExistsException, ImageCouldNotBeStoredException {
+
+        IExternalLinkDisplay externalLinkDisplay = updateLink(title, id, positionX, positionY, rotation, linkedId,
+                linkLabel, linkId, linkDisplayId, displayType, linkImage, imageFilename);
+        externalLinkDisplay.setHowToOpen(howToOpen);
```

Comment:
`updateLink` and `createLink` both save the link object I believe. The object in this line might automatically be persisted after the setter is called. I would, however, put a comment here though if that is the case (if it's not the case you have a bug here I believe).

Reply:
see above